developer/developer-guide.md

```diff
@@ -1,3 +1,872 @@
+serializes its return value to JSON.
+
+### Developer notes on search syntax
+#### Unimplemented search options:
+These options have some code towards handling, but are not implemented such that they are used in the search.
+* `facets.mincount`
+* `facets.limit`
+* `orderBy.expression`
+
+## Services
+Only the most important services are described here. Full documentation is available in 
+[the generated JavaDoc](#viewing-javadoc).
+
+### NodeService
+The `NodeService` provides operations on nodes.
+
+* Fetch and modify the metadata for a Node
+* Fetch the root node of a Store
+* Fetch, create and remove child, parent and target associations for a Node
+* Copy or move a Node to another parent
+* Create and delete Nodes
+* Checkout, checkin and fetch working copies for a Node
+
+### SearchService
+The `SearchService` allows searching for nodes based on an object tree.
+
+The `SearchService.query()` method takes a `SearchQuery` object which contains the search query to execute,
+as well as pagination, faceting and ordering options.
+
+\define{EXAMPLE_IMPORTS}
+\define{EXAMPLE_SEARCH_QUERY_OPTS}
+```java
+\include{examples/src/main/java/searchQuery.java}
+```
+\undef{EXAMPLE_IMPORTS}
+\undef{EXAMPLE_SEARCH_QUERY_OPTS}
+
+The query itself can be constructed using the `QueryBuilder`, which provides a fluent interface to 
+build search queries.
+
+\define{EXAMPLE_SEARCH_QUERY_QUERY}
+```java
+\include{examples/src/main/java/searchQuery.java}
+```
+\undef{EXAMPLE_SEARCH_QUERY_QUERY}
+
+When using the REST API, a JSON payload describing the search query has to be POST'ed to the 
+`apix/v1/search` endpoint. 
+This JSON document reflects the node structure created by the query builder, and is shown below:
+
+```json
+\include{examples/src/main/resources/jsonsearchquery.json}
+```
+
+### DictionaryService
+The `DictionaryService` provides meta-information about the metadata model.
+It allows to fetch information about registered types, aspect and properties.
+
+### Viewing JavaDoc
+Full JavaDoc documentation of the Alfred API Java interface is available in the JavaDoc. You can
+view the JavaDoc by browsing to `/alfresco/s/apix/javadocs/index.html` on your Alfresco host that
+has Alfred API installed.
+
+# Java Api
+
+The alfred-api java api the core to exposing alfresco functionality and normalizing operations across version.
+Any extensions written while depending on alfred api can be easily ported to a new alfresco version.
+
+When the api is installed, all of its service are available as beans and can be wired into your own classes.
+
+# Rest Api
+
+For a full overview of the Rest api, please refer to [the swagger specification](https://demo.xenit.eu/alfresco/s/apix/v1/docs/ui.html).
+
+## Search Requests
+
+### query
+Object containing subcomponents that build the requested query.
+All queries are translated to Alfresco Full Text Search ([AFTS]("https://community.alfresco.com/docs/DOC-5729-full-text-search-query-syntax")) by alfred-api when executed
```

Comment:
This hyperlink does not work in the PDF nor the website. Maybe due to the quotes? The above link to demo _does_ work.